Sources/StreamChat/Database/DatabaseContainer.swift

```diff
@@ -257,31 +239,20 @@ class DatabaseContainer: NSPersistentContainer {
         }
 
         writableContext.perform { [weak self] in
-            self?.sendNotificationForAllContexts(name: Self.WillRemoveAllDataNotification)
-
-            // Because this is a critical part of Stream, we don't want to make such changes unless it is in a controlled environment
-            if StreamRuntimeCheck._isBackgroundMappingEnabled {
-                // Before we remove the persistent store, we need to reset the state of the `NSManagedObjectContext`s to
-                // avoid a context from trying to access data from a previous store
-                self?.allContext.forEach { context in
-                    context.performAndWait { context.reset() }
+            let entityNames = self?.managedObjectModel.entities.compactMap(\.name)
+            var deleteError: Error?
+            entityNames?.forEach { [weak self] entityName in
+                let deleteFetch = NSFetchRequest<NSFetchRequestResult>(entityName: entityName)
+                let deleteRequest = NSBatchDeleteRequest(fetchRequest: deleteFetch)
+                do {
+                    try self?.writableContext.execute(deleteRequest)
+                    try self?.writableContext.save()
+                } catch {
+                    log.error("Batch delete request failed with error \(error)")
```

Comment:
Since there's a possibility that this operation can fail, we should also do what you suggested before - on login we check if we have existing data from another user and delete it, which should cover this failure. Wdyt?